### PR TITLE
Cknieling/remove rpk

### PR DESCRIPTION
**Pull request by chr11115**
added function to remove an rpk and it's material from a node

### CODE REVIEW

File: src/serlio/scripts/serlioCreateUI.mel [hunk 1]
```diff
@@ -60,6 +60,37 @@ global proc createPrtNode() {
 	select -r $select;
 }
 
+global proc removePrtNode() {
+	string $select[] = `ls -sl`;
+	string $initialShapes[] = collectInitialShapes($select);
+	
+	if(!(size($initialShapes))) {
```
Comment (mistafunk) [inline src/serlio/scripts/serlioCreateUI.mel:67]: for readability, please use `== 0` notation for zero size checks

File: src/serlio/scripts/serlioCreateUI.mel [hunk 2]
```diff
@@ -60,6 +60,37 @@ global proc createPrtNode() {
...
+		confirmDialog -title "Missing selection" -message "Please select shapes first in order to remove a rule." -button "OK";
```
Comment (mistafunk) [inline src/serlio/scripts/serlioCreateUI.mel:68]:
```suggestion
		confirmDialog -title "Missing selection" -message "Please select at least one shape to remove the associated rule package." -button "OK";
```

File: src/serlio/scripts/serlioCreateUI.mel [hunk 3]
```diff
@@ -101,6 +132,7 @@ global proc createPrtMenu() {
 		setParent -m $gPrtMenu;
 		menuItem -divider true;
 		menuItem -label "Attach CityEngine Rule Package..."  -c "createPrtNode" -annotation "Attach a CGA rule package to a geometry";
+		menuItem -label "Remove Rule Package Assignment"  -c "removePrtNode" -annotation "Remove a CGA rule package from a geometry";
```
Comment (mistafunk) [inline src/serlio/scripts/serlioCreateUI.mel:135]:
```suggestion
		menuItem -label "Remove CityEngine Rule Package"  -c "removePrtNode" -annotation "Remove a CGA rule package from a geometry";
```

**Review by mistafunk — CHANGES_REQUESTED** [review]
Nice work.

As discussed, we do not delete any remaining materials automatically, could have unforeseen side-effects. User can choose to use "delete unused nodes" in hypershade.

For consistency reasons, please also update these (unrelated) selection error messages in the create material scripts:
"Please select shapes first in order to create materials." -> "Please select at least one Serlio shape to create materials."

File: src/serlio/scripts/serlioCreateUI.mel (hunk 3 above)
Comment (chr11115) [reply src/serlio/scripts/serlioCreateUI.mel]: changed.

File: src/serlio/scripts/serlioCreateUI.mel (hunk 2 above)
Comment (chr11115) [reply src/serlio/scripts/serlioCreateUI.mel]: changed.

File: src/serlio/scripts/serlioCreateUI.mel (hunk 1 above)
Comment (chr11115) [reply src/serlio/scripts/serlioCreateUI.mel]: changed, also changed a few other instances with the same issue.

**Comment by chr11115** [comment]
I noticed a small bug in the code:
- "connectionInfo -dfs" returns string array, while "connectionInfo -sfd" returns string
- also added proper types to variables

File: src/serlio/scripts/AEserlioTemplate.mel [hunk 4]
```diff
@@ -62,7 +62,7 @@ global proc prtShowFileDialog(string $attr) {
 		$file = `fileDialog2 -fm 1 -cap "Select File" -ff $filters`;
 	}
 
-	if(size($file)) {  
+	if(size($file) == 0) {
```
Comment (mistafunk) [inline src/serlio/scripts/AEserlioTemplate.mel:65]: this should be `> 0`

**Review by mistafunk — CHANGES_REQUESTED** [review]
There is one typo in the size checks, afterwards feel free to merge.

File: src/serlio/scripts/AEserlioTemplate.mel (hunk 4 above)
Comment (chr11115) [reply src/serlio/scripts/AEserlioTemplate.mel]: good catch. fixed now

**Pull request merged**